.template-helpers/git-prefetch.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/env bash
+set -Eeuo pipefail
+
+repo="${1:-}"
+if [ -z "$repo" ]; then
+	echo >&2 "usage: $0 repo"
+	echo >&2 "   ie: $0 hylang"
+	exit 1
+fi
+
+# if we haven't set BASHBREW_LIBRARY explicitly (like Jenkins does, for example), don't trust the local library
+if [ -z "${BASHBREW_LIBRARY:-}" ]; then
+	repo="https://github.com/docker-library/official-images/raw/master/library/$repo"
+fi
+
+: "${BASHBREW_CACHE:=$HOME/.cache/bashbrew}"
+export BASHBREW_CACHE
+
+# see "generate-dockerfile-links-partial.tmpl" for where our arch-choosing complexity comes from
+templateHeader='{{- $archSpecific := getenv "ARCH_SPECIFIC_DOCS" -}}'
+templateArchVar='{{- $arch := $archSpecific | ternary arch ($e.HasArchitecture arch | ternary arch ($e.Architectures | first)) -}}'
+
+tags="$(bashbrew list --uniq "$repo" | cut -d: -f2-)"
+tags=( $tags )
+toFetch=()
+for tag in "$tags"; do
+	if ! bashbrew cat --format '
+		'"$templateHeader"'
+		{{- $e := $.TagEntry -}}
+		'"$templateArchVar"'
+		{{- if $e.HasArchitecture $arch -}}
+			{{- $from := $.ArchDockerFrom $arch $e -}}
+		{{- end -}}
+	' "$repo:$tag" &> /dev/null; then
+		toFetch+=( "$repo:$tag" )
+	fi
+done
+
+if [ "${#toFetch[@]}" -eq 0 ]; then
+	exit 0
+fi
+
+bashbrewFetchTemplate='
+	'"$templateHeader"'
+	{{- range $i, $e := $.Entries -}}
+		'"$templateArchVar"'
+		{{- if $e.HasArchitecture $arch -}}
+			{{- join "/" "refs/tags" $arch $.RepoName ($e.Tags | first) -}}
+			{{- ":" -}}
+			{{- "\n" -}}
+		{{- end -}}
+	{{- end -}}
+'
+
+bashbrew cat --format "$bashbrewFetchTemplate" "$repo" \
+	| xargs --no-run-if-empty git -C "$BASHBREW_CACHE/git" fetch --quiet --no-tags https://github.com/tianon/bad-ideas.git
```

Comment:
Good idea -- what do you think about https://github.com/docker-library/archive ? (that's the name we've been using for the Jenkins job that's pushing to it)

Reply:
Making it a real official thing means adding support for doing this fallback into `bashbrew` itself might even make sense.

Reply:
Updating bashbrew does indeed seem useful and would make the TravisCI test less flaky when unrelated upstream changes branches (:see_no_evil:).

The name sounds reasonable. (mostly because I cannot come up with a better name and this mostly describes what it is)

Failed name attempts:
 - remote-repo-archive
 - remote-archive
 - scrolls
 - image-chronicles
 - alexandria

Reply:
- git-archive
- manifest-archive
- commit-archive

Reply:
Kind of digging commit-archive, since that's really what it is (very explicitly).

Reply:
- warehouse

Reply:
https://github.com/docker-library/commit-warehouse :muscle: